### ISSUE TITLE
using a registry-image in a task fails in concourse 7.6.0 

### ISSUE THREAD

**Issue by natto1784** [concourse/concourse-docker#79]
Similar to #76 
stderr says
```
time="2022-02-03T08:24:16.577250855Z" level=error msg="failed to enable controllers ([cpuset cpu io memory hugetlb pids rdma])" error="failed to write subtree controllers [cpuset cpu io memory hugetlb pids rdma] to \"/sys/fs/cgroup/garden/cgroup.subtree_control\": write /sys/fs/cgroup/garden/cgroup.subtree_control: no such file or directory"
time="2022-02-03T08:24:24.862028047Z" level=info msg="starting signal loop" namespace=concourse path=/run/containerd/io.containerd.runtime.v2.task/concourse/38503fc9-0482-4038-71d8-77d6a77fd11c pid=746
time="2022-02-03T08:24:25.070572438Z" level=error msg="failed to enable controllers ([cpuset cpu io memory hugetlb pids rdma])" error="failed to write subtree controllers [cpuset cpu io memory hugetlb pids rdma] to \"/sys/fs/cgroup/cgroup.subtree_control\": write /sys/fs/cgroup/cgroup.subtree_control: operation not supported"
```
The problem is exactly the same as https://github.com/concourse/concourse/discussions/7953
Copying the entrypoint from #77 doesn't help either.  The only difference is
```diff
-time="2022-02-03T08:24:25.070572438Z" level=error msg="failed to enable controllers ([cpuset cpu io memory hugetlb pids rdma])" error="failed to write subtree controllers [cpuset cpu io memory hugetlb pids rdma] to \"/sys/fs/cgroup/cgroup.subtree_control\": write /sys/fs/cgroup/cgroup.subtree_control: operation not supported"
+time="2022-02-03T08:24:25.070572438Z" level=error msg="failed to enable controllers ([cpuset cpu io memory hugetlb pids rdma])" error="failed to write subtree controllers [cpuset cpu io memory hugetlb pids rdma] to \"/sys/fs/cgroup/cgroup.subtree_control\": write /sys/fs/cgroup/cgroup.subtree_control: device or resource busy"
```
CONCOURSE_RUNTIME is set as "containerd" for the worker.
Running this modified image https://github.com/robinhuiser/concourse-arm64 for my RPi

**Comment by robinhuiser** [comment]
Hi @natto1784!

* which version of Concourse are you using?
* is this happening with every registry image task?
* is this only applicable with dcind?
* can you share the task file?

**Comment by natto1784** [comment]
@robinhuiser Thank you for responding!
>which version of Concourse are you using?

7.6.0
> is this happening with every registry image task?

I've only tried the busybox:latest image. The task is a simple `echo hello world`
>is this only applicable with dcind?

Likely, but I have only used the docker image for concourse (https://hub.docker.com/r/rdclda/concourse) since I'm running it on a Raspberry PI 4. Please note that I'm running two concourse images separately as web and worker nodes, both under Nomad. 
>can you share the task file?

[https://git.weirdnatto.in/natto1784/pipelines/src/branch/master/parent.yml](url)


The web UI shows this btw
![Concourse UI](https://user-images.githubusercontent.com/56316606/152313884-d2d13178-62e0-430b-8442-7ad354ffb98e.png)

**Comment by natto1784** [comment]
Okay, this solved itself, looks like I wasted so many hours over nothing. What I did was
1. Wipe the entire concurse-worker work directory
2. WIpe the entire concourse DB (it was a fresh install anyway)
3. Set "default-cgroupns-mode" to "host" in docker daemon

Then it worked! With no errors!

4. Then I removed the "default-cgroupns-mode" setting I had set earlier

After restarting everything, it worked this time too! But the errors I mentioned in my original post are still there, I did read up on cgroups, but I'm not sure how these errors can affect me presently. At least, everything works now.
Another thing to note is that I did NOT use the patch in #77 and it still worked. The patch still has the same effect that I mentioned in my original post.
